File: Example/NXNavigationExtensionDemo/Feature/Basic/Controllers/ViewController13_Blur.m
//
//  ViewController13_Blur.m
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2021/5/4.
//

#import <NXNavigationExtension/NXNavigationExtension.h>

#import "ViewController13_Blur.h"
#import "UIColor+RandomColor.h"

@interface ViewController13_Blur ()

@property (nonatomic, strong) UIVisualEffectView *gaussianBlurView;

@end

@implementation ViewController13_Blur

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.nx_navigationBar addSubview:self.gaussianBlurView];
    
    [self.gaussianBlurView.topAnchor constraintEqualToAnchor:self.nx_navigationBar.topAnchor].active = YES;
    [self.gaussianBlurView.leftAnchor constraintEqualToAnchor:self.nx_navigationBar.leftAnchor].active = YES;
    [self.gaussianBlurView.bottomAnchor constraintEqualToAnchor:self.nx_navigationBar.bottomAnchor].active = YES;
    [self.gaussianBlurView.rightAnchor constraintEqualToAnchor:self.nx_navigationBar.rightAnchor].active = YES;
}

- (UIColor *)nx_navigationBarBackgroundColor {
    return [UIColor clearColor];
}

#pragma mark - Getter & Setter

- (UIVisualEffectView *)gaussianBlurView {
    if (!_gaussianBlurView) {
        UIBlurEffect *effect;
        if (@available(iOS 13.0, *)) {
            effect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleSystemChromeMaterial];
        } else {
            effect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleExtraLight];
        }
        _gaussianBlurView = [[UIVisualEffectView alloc] initWithEffect:effect];
        _gaussianBlurView.translatesAutoresizingMaskIntoConstraints = NO;
        
        UIView *contentView = [[UIView alloc] init];
        contentView.translatesAutoresizingMaskIntoConstraints = NO;
        contentView.backgroundColor = [UIColor customColorWithLightModeColor:^UIColor * _Nonnull{
            return [UIColor redColor];
        } darkModeColor:^UIColor * _Nonnull{
            return [UIColor grayColor];
        }];
        [_gaussianBlurView.contentView addSubview:contentView];
        [_gaussianBlurView.contentView setAlpha:0.5];
        
        [contentView.topAnchor constraintEqualToAnchor:self.gaussianBlurView.topAnchor].active = YES;
        [contentView.leftAnchor constraintEqualToAnchor:self.gaussianBlurView.leftAnchor].active = YES;
        [contentView.bottomAnchor constraintEqualToAnchor:self.gaussianBlurView.bottomAnchor].active = YES;
        [contentView.rightAnchor constraintEqualToAnchor:self.gaussianBlurView.rightAnchor].active = YES;
    }
    return _gaussianBlurView;
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController13_Blur.swift
//
//  ViewController13_Blur.swift
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2021/5/7.
//

import UIKit

class ViewController13_Blur: BaseTableViewController {
    
    private lazy var blurView: UIVisualEffectView = {
        var effect: UIBlurEffect! = nil
        if #available(iOS 13.0, *) {
            effect = UIBlurEffect(style: .systemChromeMaterial)
        } else {
            effect = UIBlurEffect(style: .extraLight)
        }
        
        let blurView = UIVisualEffectView(effect: effect)
        blurView.translatesAutoresizingMaskIntoConstraints = false
        
        let contentView = UIView()
        contentView.translatesAutoresizingMaskIntoConstraints = false
        contentView.backgroundColor = UIColor.customDynamic(lightMode: { .red }, darkMode: { .gray })
        blurView.contentView.addSubview(contentView)
        blurView.contentView.alpha = 0.5
        
        contentView.topAnchor.constraint(equalTo: blurView.topAnchor).isActive = true
        contentView.leftAnchor.constraint(equalTo: blurView.leftAnchor).isActive = true
        contentView.bottomAnchor.constraint(equalTo: blurView.bottomAnchor).isActive = true
        contentView.rightAnchor.constraint(equalTo: blurView.rightAnchor).isActive = true
        
        return blurView
    }()

    override func viewDidLoad() {
        super.viewDidLoad()

        nx_navigationBar.addSubview(blurView)
        
        blurView.topAnchor.constraint(equalTo: nx_navigationBar.topAnchor).isActive = true
        blurView.leftAnchor.constraint(equalTo: nx_navigationBar.leftAnchor).isActive = true
        blurView.bottomAnchor.constraint(equalTo: nx_navigationBar.bottomAnchor).isActive = true
        blurView.rightAnchor.constraint(equalTo: nx_navigationBar.rightAnchor).isActive = true
    }
    
    override var nx_navigationBarBackgroundColor: UIColor? {
        return .clear
    }

}
